app/controllers/coronavirus_form/data_export_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class CoronavirusForm::DataExportController < ApplicationController
+  def show
+    render plain: produce_csv(usage_statistics)
+  end
+
+private
+
+  def usage_statistics
+    questions = I18n.t("coronavirus_form.groups").map { |_, v| v[:questions] }.reduce(:merge)
+
+    start_date = Date.parse(sanitize(params[:start_date])).beginning_of_day
+    end_date = Date.parse(sanitize(params[:end_date])).end_of_day
+
+    results = {}
+    questions.each_key do |question|
+      question_text = questions.dig(question, :title)
+      counts = FormResponse.
+        where(created_at: start_date..end_date).
+        select(%{created_at::date, form_response -> '#{question}', COUNT(*)}).
+        group(%{created_at::date, form_response -> '#{question}'}).
+        pluck(%{created_at::date, form_response -> '#{question}', COUNT(*)}).
+        reject { |v| v[1].nil? }.
+        map do |v|
+          {
+            date: v[0].strftime("%Y-%m-%d"),
+            response: Array(v[1]).join(";"),
+            count: v[2],
+          }
+        end
+      results[question_text] = counts
+    end
+    results
+  end
+
+  def produce_csv(results)
```

Comment:
Ruby has a CSV class that could be worth investigating: https://ruby-doc.org/stdlib-2.6.1/libdoc/csv/rdoc/CSV.html

Reply:
Thanks, I'll use this.